programs/uxd/src/instructions/credix_lp/rebalance_request_execute_from_credix_lp_depository.rs

```diff
@@ -0,0 +1,646 @@
+    // -- Phase 3
+    // -- Compute the profits and the overflow of the depository we will want to withdraw
+    // ---------------------------------------------------------------------
+
+    let redeemable_amount_under_management = checked_convert_u128_to_u64(
+        ctx.accounts
+            .depository
+            .load()?
+            .redeemable_amount_under_management,
+    )?;
+
+    let profits_collateral_amount = owned_shares_value_before
+        .checked_sub(redeemable_amount_under_management)
+        .ok_or(UxdError::MathError)?;
+
+    let overflow_value = {
+        let redeemable_amount_under_management_target_amount =
+            calculate_credix_lp_depository_target_amount(
+                &ctx.accounts.controller,
+                &ctx.accounts.identity_depository,
+                &ctx.accounts.mercurial_vault_depository,
+                &ctx.accounts.depository,
+            )?;
+        if redeemable_amount_under_management < redeemable_amount_under_management_target_amount {
+            0
+        } else {
+            redeemable_amount_under_management
+                .checked_sub(redeemable_amount_under_management_target_amount)
+                .ok_or(UxdError::MathError)?
+        }
+    };
+
+    // ---------------------------------------------------------------------
+    // -- Phase 4
+    // -- Here we compute how much we would like to withdraw now,
+    // -- We also need to compute how much credix will let us withdraw now,
+    // -- And where the withdrawn collateral will be going (profits or rebalanced)
+    // ---------------------------------------------------------------------
+
+    // Read credix withdrawal accounts onchain state
+    let locked_liquidity = ctx.accounts.credix_global_market_state.locked_liquidity;
+    let investor_total_lp_amount = ctx
+        .accounts
+        .credix_withdraw_request
+        .investor_total_lp_amount;
+    let participating_investors_total_lp_amount = ctx
+        .accounts
+        .credix_withdraw_epoch
+        .participating_investors_total_lp_amount;
+    let base_amount_withdrawn = ctx.accounts.credix_withdraw_request.base_amount_withdrawn;
+
+    // All investors gets an equivalent slice of the locked liquidity,
+    // based on their relative position size in the lp pool
+    let withdrawable_total_collateral_amount = locked_liquidity
+        .checked_mul(investor_total_lp_amount)
+        .ok_or(UxdError::MathError)?
+        .checked_div(participating_investors_total_lp_amount)
+        .ok_or(UxdError::MathError)?
+        .checked_sub(base_amount_withdrawn)
+        .ok_or(UxdError::MathError)?;
+
+    // We prioritize withdrawing the overflow first based on what liquidity is available
+    let withdrawal_overflow_value =
+        std::cmp::min(withdrawable_total_collateral_amount, overflow_value);
+    // Then if liquidity is still available for profits, also withdraw it
```

Comment:
Note: we should discuss this one with Kento. Should profit be prioritize versus balance? Especially considering profits represents way smaller number.

Reply:
I think I asked this a while back in the slack channel, we can discuss again, but basically the reasoning at the time was:
- an imbalance represent additional risks or insuficient risks
- so an imbalance is more critical than paying out the UXP holders immediately

Reply:
Discussed by voice chat, team voted to prioritize profits instead of balance. Commited the change!